### PR TITLE
Add changelog for COOAssembler

### PR DESCRIPTION
**Pull request by KnutAM**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -456,6 +456,8 @@ more discussion).
 - The function `reshape_to_nodes` have been deprecated in favor of `evaluate_at_grid_nodes`.
   ([#703])
 
+- `start_assemble([n::Int])` has been deprecated in favor of calling `COOAssembler()` directly ([#916], [#1058]).
```
Comment (fredrikekre) [inline CHANGELOG.md:459]:
```suggestion
- `start_assemble([n::Int])` has been deprecated in favor of calling `Ferrite.COOAssembler()` directly ([#916], [#1058]).
```

Comment (fredrikekre) [reply CHANGELOG.md:459]: Maybe also make it an error in deprecated.jl for now?

File: src/deprecations.jl [hunk 2]
```diff
@@ -443,3 +443,6 @@ end
 function assemble!(::AbstractAssembler, ::AbstractVector{<:Integer}, ::AbstractVector, ::AbstractMatrix)
     throw(DeprecationError("assemble!(assembler, dofs, fe, Ke)" => "assemble!(assembler, dofs, Ke, fe)"))
 end
+
+start_assemble(::Int) = throw(DeprecationError("start_assemble(n::Int)" => "Ferrite.COOAssembler(nrows, ncols)"))
+start_assemble() = throw(DeprecationError("start_assemble(n::Int)" => "Ferrite.COOAssembler()"))
```
Comment (fredrikekre) [inline src/deprecations.jl:448]:
```suggestion
start_assemble() = throw(DeprecationError("start_assemble()" => "Ferrite.COOAssembler()"))
```

Comment (KnutAM) [reply src/deprecations.jl]: Thanks - was a bit too fast :)

File: src/deprecations.jl [hunk 3]
```diff
@@ -443,3 +443,9 @@ end
 function assemble!(::AbstractAssembler, ::AbstractVector{<:Integer}, ::AbstractVector, ::AbstractMatrix)
     throw(DeprecationError("assemble!(assembler, dofs, fe, Ke)" => "assemble!(assembler, dofs, Ke, fe)"))
 end
+
+start_assemble(::Int) = throw(DeprecationError("start_assemble(n::Int)" => "Ferrite.COOAssembler(nrows, ncols)"))
```
Comment (fredrikekre) [inline src/deprecations.jl:447]:
```suggestion
start_assemble(::Int) = throw(DeprecationError("start_assemble(n::Int)" => "Ferrite.COOAssembler(; sizehint = n)"))
```

**Pull request merged**